internal/gps/result.go

```diff
@@ -65,37 +64,59 @@ func WriteDepTree(basedir string, l Lock, sm SourceManager, sv bool, logger *log
 		return err
 	}
 
-	var wg sync.WaitGroup
-	errCh := make(chan error, len(l.Projects()))
+	lps := l.Projects()
+
+	type resp struct {
+		i   int
+		err error
+	}
+	respCh := make(chan resp, len(lps))
+
+	for i := range lps {
+		go func(i int) {
+			p := lps[i]
+
+			var err error
+			defer func() {
+				if r := recover(); r != nil {
```

Comment:
I haven't seen one, but I wanted to be careful not to return a false positive if one occurs.